test/e2e/preemption/preemption_test.go

```diff
@@ -0,0 +1,321 @@
+	Ω(nodesDAOInfo).NotTo(gomega.BeNil())
+
+	for _, node := range *nodesDAOInfo {
+		if node.NodeID == Worker {
+			WorkerMemRes = node.Available["memory"]
+		}
+	}
+	WorkerMemRes /= (1000 * 1000) // change to M
+	fmt.Fprintf(ginkgo.GinkgoWriter, "Worker node %s available memory %dM\n", Worker, WorkerMemRes)
+
+	sleepPodMemLimit = int64(float64(WorkerMemRes) / 3)
+	Ω(sleepPodMemLimit).NotTo(gomega.BeZero(), "Sleep pod memory limit cannot be zero")
+	fmt.Fprintf(ginkgo.GinkgoWriter, "Sleep pod limit memory %dM\n", sleepPodMemLimit)
+})
+
+var _ = ginkgo.AfterSuite(func() {
+
+	ginkgo.By("Untainting some nodes")
+	err := kClient.UntaintNodes(nodesToTaint, taintKey)
+	Ω(err).NotTo(gomega.HaveOccurred(), "Could not remove taint from nodes "+strings.Join(nodesToTaint, ","))
+
+	ginkgo.By("Check Yunikorn's health")
+	checks, err := yunikorn.GetFailedHealthChecks()
+	Ω(err).NotTo(gomega.HaveOccurred())
+	Ω(checks).To(gomega.Equal(""), checks)
+
+	testDescription := ginkgo.CurrentSpecReport()
+	if testDescription.Failed() {
+		tests.LogTestClusterInfoWrapper(testDescription.FailureMessage(), []string{ns.Name})
+		tests.LogYunikornContainer(testDescription.FailureMessage())
+	}
+	ginkgo.By("Tearing down namespace: " + ns.Name)
+	err = kClient.TearDownNamespace(ns.Name)
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	yunikorn.RestoreConfigMapWrapper(oldConfigMap, annotation)
+})
+
+var _ = ginkgo.Describe("Preemption", func() {
+	ginkgo.It("Verify_basic_preemption", func() {
+		ginkgo.By("A queue uses resource more than the guaranteed value even after removing one of the pods. The cluster doesn't have enough resource to deploy a pod in another queue which uses resource less than the guaranteed value.")
+		// update config
+		ginkgo.By("Update root.sandbox1 and root.sandbox2 with guaranteed memory " + strconv.FormatInt(sleepPodMemLimit, 10) + "M")
+		cm, err := kClient.GetConfigMap(constants.ConfigMapName, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err).NotTo(HaveOccurred())
+		cm.Data[configmanager.DefaultPolicyGroup] = fmt.Sprintf(preemptionQueueYamlFormat, sleepPodMemLimit, "default", sleepPodMemLimit)
+		_, err = kClient.UpdateConfigMap(cm, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err).NotTo(HaveOccurred())
+
+		// Define sleepPod
+		sleepPod1Configs := k8s.SleepPodConfig{Name: "sleepjob1", NS: dev, Mem: sleepPodMemLimit, Time: 600, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod2Configs := k8s.SleepPodConfig{Name: "sleepjob2", NS: dev, Mem: sleepPodMemLimit, Time: 600, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod3Configs := k8s.SleepPodConfig{Name: "sleepjob3", NS: dev, Mem: sleepPodMemLimit, Time: 600, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod4Configs := k8s.SleepPodConfig{Name: "sleepjob4", NS: dev, Mem: sleepPodMemLimit, Time: 600, Optedout: true, Labels: map[string]string{"queue": "root.sandbox2"}}
```

Comment:
Pod initialization is almost the same in every test, except the "Time" part, think about extracting this to a function.